core/libp2p/multi/multibase_codec/codecs/base16.cpp

```diff
@@ -40,28 +49,24 @@ namespace libp2p::multi::detail {
     return hex_lower(bytes.toVector());
   }
 
-  Result<Buffer, std::string> decodeBase16Upper(std::string_view string) {
+  outcome::result<Buffer> decodeBase16Upper(std::string_view string) {
     // we need this check, because Boost can unhex any kind of base16 with one
     // func, but the base must be specified correctly
     if (!encodingCaseIsUpper(string)) {
-      return Error{"cannot unhex string '" + std::string{string}
-                   + "': input is not in the uppercase hex"};
+      return Base16DecodeError::kNonUppercaseInput;
     }
-    return unhex(string) | [](auto &&v) -> Result<Buffer, std::string> {
-      return Value{Buffer{std::forward<decltype(v)>(v)}};
-    };
+    OUTCOME_TRY(bytes, unhex(string));
+    return Buffer{std::move(bytes)};
   }
 
-  Result<Buffer, std::string> decodeBase16Lower(std::string_view string) {
+  outcome::result<Buffer> decodeBase16Lower(std::string_view string) {
     // we need this check, because Boost can unhex any kind of base16 with one
     // func, but the base must be specified correctly
     if (encodingCaseIsUpper(string)) {
-      return Error{"cannot unhex string '" + std::string{string}
-                   + "': input is not in the lowercase hex"};
+      return Base16DecodeError::kNonLowercaseInput;
     }
-    return unhex(string) | [](auto &&v) -> Result<Buffer, std::string> {
-      return Value{Buffer{std::forward<decltype(v)>(v)}};
-    };
+    OUTCOME_TRY(bytes, unhex(string));
+    return Buffer{std::move(bytes)};
```

Comment:
unhex returns std::vector, and Buffer can be implicitly created from vector, so simply ```return unhex(string);```

Reply:
I believe that the constructor from vector is marked explicit.

Reply:
you can remove `explicit` and return `unhex(string);`